1.6/contrib/configuration/jobs/OpenShift Sample/config.xml

```diff
@@ -0,0 +1,153 @@
+    </hudson.model.ParametersDefinitionProperty>
+  </properties>
+  <scm class="hudson.scm.NullSCM"/>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>
+if [ -z "$CREDENTIALS" ]; then
+  SERVER=https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}
+  TOKEN=`cat /var/run/secrets/kubernetes.io/serviceaccount/token`
+  oc config set-cluster local-cluster --server=$SERVER --insecure-skip-tls-verify=true
+  oc config set-credentials local-user --token=$TOKEN
+  oc config set-context local --cluster=local-cluster --user=local-user --namespace=$PROJECT
+  oc config use-context local
+else 
+  export KUBECONFIG=.kubeconfig
+  echo "${CREDENTIALS}" > "${KUBECONFIG}"
+fi
+
+oc project ${PROJECT}
+
+# Switch back to .clusterIP once there is a new openshift release where we can get the updated oc binary 
+#TEST_ENDPOINT=`oc get service ${SERVICE} -t '{{.spec.clusterIP}}{{":"}}{{ $a:= index .spec.ports 0 }}{{$a.port}}'`
+TEST_ENDPOINT=`oc get service ${SERVICE} -t '{{.spec.portalIP}}{{":"}}{{ $a:= index .spec.ports 0 }}{{$a.port}}'`
+
+echo "Scaling down old test deployments"
+oc get rc -t '{{ range .items }}{{.spec.selector.deploymentconfig}}{{" "}}{{.metadata.name}}{{"\n"}}{{end}}' | grep -e "^$DEPLOYMENT_CONFIG " | awk '{print $2}' | while read -r test_rc_id; do
+  echo "Scaling down old deployment $test_rc_id"
+  oc scale --replicas=0 rc $test_rc_id
+done
+
+# wait for old pods to be torn down
+sleep 5
+
+echo "Triggering new application build and deployment"
+BUILD_ID=`oc start-build ${BUILD_CONFIG}`
+
+# stream the logs for the build that just started
+oc build-logs $BUILD_ID
+
+
+echo "Checking build result status"
+rc=1
+count=0
+attempts=100
+while [ $rc -ne 0 -a $count -lt $attempts ]; do
+  status=`oc get build ${BUILD_ID} -t '{{.status.phase}}'`
+  if [[ $status == "Failed" || $status == "Error" || $status == "Canceled" ]]; then
+    echo "Fail: Build completed with unsuccessful status: ${status}"
+    exit 1
+  fi
+
+  if [ $status == "Complete" ]; then
+    echo "Build completed successfully, will test deployment next"
+    rc=0
+  else 
+    count=$(($count+1))
+    echo "Attempt $count/$attempts"
+    sleep 5
+  fi
+done
+
+if [ $rc -ne 0 ]; then
+    echo "Fail: Build did not complete in a reasonable period of time"
+    exit 1
+fi
+
+# scale up the test deployment
+oc get rc -t '{{ range .items }}{{.spec.selector.deploymentconfig}}{{" "}}{{.metadata.name}}{{"\n"}}{{end}}' | grep -e "^$DEPLOYMENT_CONFIG " | awk '{print $2}' | while read -r test_rc_id; do
+  # at the end of this loop, 'test_rc_id' will contain the id of the last deployment in the list
+  # should be the most recent.
+  echo $test_rc_id > test_rc_id
+done
+
+test_rc_id=`cat test_rc_id`
```

Comment:
tl;dr: test_rc_id is only in scope within the loop and there's no simple way to export it out of that scope because it's actually a separate process.